src/mscorlib/src/System/DateTime.cs

```diff
@@ -781,9 +781,9 @@ private static long TimeToTicks(int hour, int minute, int second)
             // and y100 are relative to year 1, not year 0
             bool leapYear = y1 == 3 && (y4 != 24 || y100 == 3);
             int[] days = leapYear? DaysToMonth366: DaysToMonth365;
-            // All months have less than 32 days, so n >> 5 is a good conservative
-            // estimate for the month
-            int m = n >> 5 + 1;
+			// All months have less than 32 days so n >> 5, which is the same as n / 32,
+			// is a good conservative estimate for the month
+			int m = (n >> 5) + 1;
```

Comment:
The logic still works fine with dividing by 64 and m being 0 as far as I can tell. The only side effect is extra iterations over days[m]. I submitted the pull request because the logic was inconsistent with what the comments were saying and since you're performing the calculations you might as well save the iterations over days[m].

To specifically address your point if n == 0 and m == 0
```C#
while (n >= days[m]) m++;
```
will be reduced to
```C#
while (0 >= days[0]) m++;
```
which reduces further to
```C#
while (0 >= 0) m++;
```
which means m is always at least 1.